extensions/pkg/terraformer/terraformer.go

```diff
@@ -263,18 +263,25 @@ func (t *terraformer) execute(ctx context.Context, command string) error {
 			}
 		}
 
-		podLogger.Info("Cleaning up Terraformer pod")
-		// If the context error is non-nil (cancelled or deadline exceeded),
-		// create a new context for deleting the pod since attempting to use the original context will fail.
-		// The context might get cancelled for example by the owner check watchdog and the pod should be
-		// deleted to prevent the terraform script from running after the context was cancelled.
-		if ctx.Err() != nil {
+		switch {
+		case errors.Is(ctx.Err(), context.Canceled):
+			// If the context error is Canceled, the parent context has been canceled. Because the Terraform is fairly unstable
+			// and interruptions may cause it to not properly store the state, we will allow it to continue. The next reconciliation will
+			// adopt the running pod.
+			podLogger.Info("Skipping Terraformer pod deletion because context was cancelled")
+		case errors.Is(ctx.Err(), context.DeadlineExceeded):
+			// If the context error is deadline exceeded, create a new context for deleting the pod since attempting to use the
+			// original context will fail. The context might get cancelled for example by the owner check watchdog and the pod
```

Comment:
We should no longer have any owner check watchdogs, do we?

Reply:
You are right. Owner check + owner check watchdog has been removed as part of bad case control plane migration removal - ref https://github.com/gardener/gardener/issues/6302.